### PR TITLE
docs: Cancun hip design document

### PR DESCRIPTION
**Pull request by shemnon**
**Description**:

Design document covering the HIPs for Cancun support in the EVM.

**Related issue(s)**:

documents #10272

**Notes for reviewer**:

Just a design doc

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# Cancun Fork Support
+
+## Purpose
+
+There Ethereum "Cancun" hard introduces a number of changes to the EVM that will need to be implemented to maintain EVM
```
Comment (Nana-EC) [inline hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md:5]:
```suggestion
The Ethereum "Cancun" hardfork introduces a number of changes to the EVM that will need to be implemented to maintain EVM
```

File: hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+Equivalence. There are four different HIPs covering this feature (HIP-865: Add EVM Support for transient storage and
```
Comment (Nana-EC) [inline hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md:6]: Should turn this into a list 
```suggestion
Equivalence. There are four different HIPs covering this feature 
- HIP-865: Add EVM Support for transient storage
- HIP-866: Add EVM compatibility for non-supported Cancun blob features
- HIP-867: Add Smart Contract Services Support for KZG Point Evaluation Precompiled Function
- HIP-868: Support Cancun Self-Destruct
```

File: hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+how the HIPs will be implemented in Hedera Services.
+
+Generally speaking there are four strategies that will be adopted:
+
+1. Adopt without change. This includes the `MCOPY`, `TSTORE`, and `TLOAD` operations as well as the KZG precompile.
+2. Adapt with Hedera Considerations. This includes the `SELFDESTRUCT` changes.
+3. Do not adopt but stub out placeholders. This includes the `VERSIONEDHASH` and `BLOBBASEFEE` operations.
+4. Do not adopt. This includes all Blob handling and the "Type 3" blob transaction.
+
+### Example User Story
```
Comment (Nana-EC) [inline hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md:19]: Let's have more specific user stories highlighting the capabilities to users

File: hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+## Non Goals
+
+It is not a goal to introduce blobs into Hedera. It is also not a goal to restrict design space nor to dictate future
+design directions for support or non-support of blobs.
+
+## Implementation
+
+### New Cancun EVM
+
+A new EVM version will be created that will be based off of the Cancun EVM. Simply activating the Cancun EVM should get
```
Comment (Nana-EC) [inline hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md:48]: Cna we be explicit on what activating means? E.g. what config needs to be xhanged and to what?

File: hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+* `MCOPY` operation
+* `TLOAD` and `TSTORE` operations
+* `VERSIONEDHASH` and `BLOBBASEFEE` Operations that query the `TxValues` Object
+* KZG Precompiled contract
+
+### Set Correct Values in TxValues
+
+There are two new fields in the `TxValues` object that need to be set correctly. We need to ensure
+that `versionedHashes` is st to an empty list and that `blobGasPrice` is set to one. We should be able to do that in all
```
Comment (Nana-EC) [inline hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md:59]:
```suggestion
that `versionedHashes` is set to an empty list and that `blobGasPrice` is set to one. We should be able to do that in all
```

Comment (Nana-EC) [inline hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md:59]: Set them in the EVM versions how? Is there a specific class that needs to be updated or a a method that needs to be overrided?

File: hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+EVM versions so setting them unconditionally in the `MessageFrame` construction in `HederaEvmTxProcessor` should be
+sufficient.
+
+### Update HederaSelfDestructOperation behavior
+
+Either with a new class or a class that takes a constructor parameter, update the HederaSelfDestructOperation so that it
```
Comment (Nana-EC) [inline hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md:65]: Please specify which option the implementation should follow. Feel free to not that the other wasn't chosen for reason X

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looking good.
I think a bit more detail would be useful.
The design docs value is that any dev should be able to pick it up and go ahead and implement, plus reviewers of PRs should be able to reference this and map the changes to the intentions here.

**Comment by david-bakin-sl** [comment]
The commit I just pushed addresses _some_ PR review comments; not all.  The remaining comments are actually _inline_ in the document as HTML comments - this is to facilitate my updating the document in my repo clone.  They'll be properly removed when they're addressed.

**Comment by shemnon** [comment]
Closing as stale.  If needed @david-bakin-sl  or @Nana-EC can re-create the PR.

**Comment by Nana-EC** [comment]
> Closing as stale. If needed @david-bakin-sl or @Nana-EC can re-create the PR.

Thanks @shemnon. Cancun support should be going out in Hedera 0.50.0, appreciate the leadership and contribution here.

**Pull request closed without merging**